gammapy/utils/scripts.py

```diff
@@ -133,23 +136,66 @@ def _configure_root_logger(level='info', format=None):
 
 
 def read_yaml(filename, logger=None):
-    """Read config from YAML file."""
+    """
+    Read config from YAML file 
+    """
     import yaml
     if logger is not None:
         logger.info('Reading {}'.format(filename))
     with open(filename) as fh:
         config = yaml.safe_load(fh)
+
     return config
 
 
 def write_yaml(config, filename, logger=None):
-    """Write YAML config file
+    """
+    Write YAML config file
 
     This function can be used by scripts that alter the users config file.
+
+    Parameters
+    ----------
+    config : dict
+        config to write
+    filename : str, `~gammapy.exter.pathlib.Path`
+        file to write
     """
     import yaml
-    filename = filename + '.yaml'
+    filename = make_path(filename)
     if logger is not None:
         logger.info('Writing {}'.format(filename))
-    with open(filename, 'w') as outfile:
+    with open(str(filename), 'w') as outfile:
         outfile.write(yaml.dump(config, default_flow_style=False))
+
+
+def make_path(path):
+    """
+    Expand environment varibles on `~pathlib.Path` construction
+
+    Parameters
+    ----------
+    path : str, `~gammapy.extern.pathlib.Path`
+        path to expand
+    """
+    return Path(expandvars(str(path)))
+
+
+def recursive_update(old, new):
```

Comment:
How is this different from plain `dict.update`?
https://docs.python.org/2/library/stdtypes.html#dict.update

Even if it is different, there's probably a way to do it directly with dict or one of the existing containers / functions in the Python standard library, and if so I'd like to avoid us introducing a dict helper function.

If it's not clear how to do it elegantly without a helper function, can you drop by and we talk about this in person?